components/bl-PhotoSwipeGallery/bundle.js

```diff
@@ -0,0 +1,144 @@
+    const images = data.map(imgData => {
+      const imageRelation = relations.find(el => el.url === imgData.URL);
+      return React.createElement(
+        'a', 
+        {
+          href: imgData.URL,
+          'data-pswp-width': 3840,
+          'data-pswp-height': 3840 / imageRelation.relation,
+          key: `photoSwipeGallery-${imgData.objectId}`,
+          target: "_blank",
+          rel: "noreferrer",
+        },
+        React.createElement(
+          'img', {
+          src: imgData.URL,
+          alt: "",
+          style: {
+            width: "15vw",
+            height: "auto"
+          }
+          }
+        )
+      )
+    })
+    
+    return images;
+  }
+  
+  function createRelations(data) {
+    return data.map(img => {
+      const size = {};
+      size.url = img.URL;
+      const image = new Image();
+      image.src = img.URL;
+      size.relation = image.width / image.height;
+      return size;
+    })
+  }
+  
+  function createGallery(data, show=true) {
+    const relations = createRelations(data);
+    const images = createImages(data, relations);
+    const display = show? "flex" : "none"
+    const gallery = React.createElement(
+      'div', 
+      {
+        className: 'pswp-gallery',
+        id: 'photoSwipeGallery',
+        style: {display:display, 'flex-wrap': 'wrap'},
+      },
+      images
+    )
+    
+    return gallery;
+  }
+
+  
+  return function CustomComponent({component, eventHandlers}) {
+    const [galleryComponent, setGalleryComponent] = React.useState(null);
+    const elRef = React.useRef(null)
+    
+    
+    
+    React.useEffect(() => {
+      
+      if(component.imagesUrlLogic) {
+        const newData = Object.assign({}, component.imagesUrlLogic);
+        const imgDataArray = []; 
+        
+        for(const key in newData) {
+          const point = Object.assign({},newData[key]);
+          imgDataArray.push(point);
+        }
+        
+        if (component.display) {
+          setGalleryComponent(createGallery(imgDataArray));
+        } else {
+          setGalleryComponent(createGallery(imgDataArray, false));
+        }
+        
+        const options = {
+          bgOpacity: Number(component.bgOpacity),
+          separator: component.separator,
+          allowPanToNext: component.allowPanToNext === 'true',
+          loop: component.loop === 'true',
+          wheelToZoom: component.wheelToZoom === 'true',
+          pinchToClose: component.pinchToClose === 'true',
+          closeOnVerticalDrag: component.closeOnVerticalDrag === 'true',
+          escKey: component.escKey === 'true',
+          arrowKeys: component.arrowKeys === 'true',
+        }
+        
+        initLibrary(options);
+        
+      }
+      
+    },[component.imagesUrlLogic, component.imagesUrlLogic?.length, component.display]);
+    
+    return React.createElement('div', {className: "bl-customComponent-photoGallery", ref: elRef}, galleryComponent);
```

Comment:
need to rework the component with using a new approach with JSX and build system